src/bundle/Resources/public/js/scripts/core/dropdown.js

```diff
@@ -160,7 +165,12 @@
             }
 
             this.fitItems();
-            this.fireValueChangedEvent();
+
+            if (this.currentSelectedValue !== value) {
+                this.fireValueChangedEvent();
+
+                this.currentSelectedValue = value;
```

Comment:
nitpick, I would expect `currentSelectedValue` to be set in before firing event.
```suggestion
                this.currentSelectedValue = value;

                this.fireValueChangedEvent();
```